internal/kubernetes/operator/dbusers/dbusers.go

```diff
@@ -70,10 +70,13 @@ func BuildDBUsers(provider store.OperatorDBUsersStore, projectID, projectName, t
 				},
 			},
 			Spec: akov2.AtlasDatabaseUserSpec{
-				Project: akov2common.ResourceRefNamespaced{
+				Project: &akov2common.ResourceRefNamespaced{
 					Name:      resources.NormalizeAtlasName(projectName, dictionary),
 					Namespace: targetNamespace,
 				},
+				ExternalProjectRef: &akov2.ExternalProjectReference{
+					ID: projectID,
+				},
```

Comment:
The flow of building an invalid CR and later fixing it seems counterintuitive to me. Why have you followed this approach?

Reply:
Here we have the information to generate both references, but not the context to know which one is required.

I could revisit, but it requires to update the generator interface and pass an extra argument, the `independentResources` flag.

I wanted to write something that would capture the blunt of the change for all resources, so adding support going forward was as simple as adding a type to a list, or a call to a swicht/case, rather than copying and pasting very similar code again and again.

I see the issue, I will try to find another way.

Reply:
I went for the simplest case, which will mean passing the flag every time we need it. It will not be declarative.